### ISSUE TITLE
Idea – Measure users that blocked GTM and/or GA

### ISSUE THREAD

**Issue by erikmolenaarnl** [duracelltomi/gtm4wp#245]
Hi,

In this privacy-first world with increasing usage of adblockers and browsers blocking GTM containers and GA4 events, where server-side GTM is still both costly and lots of hassle, I might have a great idea!

The idea of below article could be implemented within GTM4WP, so it shows stats how many visitors were blocking GTM from the WP Dashboard.

https://www.diogo.in/en/how-to-measure-users-that-blocked-google-analytics-and-google-tag-manager-in-a-private-way/

GTM4WP would get an extra tab with percentages of blocked/succesful GTM containers (and possibly GA).

I am not a great developer myself, but ny team could develop this and commit for the community, but I want to check first what your thoughts are and thus the probability a feature like this will be included in core. What do you think of this idea?

Looking forward to your thoughts. Thanks :-)

**Comment by duracelltomi** [comment]
Hi,

Thanks for the idea!

I would say this could be a great addition, but it needs to be a feature that can be turned on and off and it must be turned off by default.

Also, my biggest concern is always about performance. I do not think this needs to be tracked on all page loads in a session for example. When a user is blocking anything, it will be blocked on all page loads, there is no need to store this information multiple times in a session.

And I would say storing browser name and major/minor version should be enough to prevent any chance of fingerprinting.

**Comment by erikmolenaarnl** [comment]
Thanks for your thoughts.

It does need some more pondering, as

A) the GTM container can be blocked

but if it is not

B) its tag events within the container (GA3/GA4/Meta/etc) can subsequently still be blocked.

So reporting should be GTM as well as the individual tag events. I am sure this is achievable within a plugin and/or client-side tracking.

Your thoughts are very welcome in fleshing this concept out! Thanks.

**Comment by duracelltomi** [comment]
Thanks again for this idea, @erikmolenaarnl — and for the thoughtful back-and-forth back in 2022. Coming back to it, I've decided not to build it into GTM4WP, for two reasons:

1. **Reliability.** You can't reliably measure what's blocked: the ad/tracker blockers that block your GA/GTM *tags* frequently block the GTM *container* itself — and any measurement script you'd add — at higher blocking levels. So the "blocked" visitors are largely the same population you can't measure in the first place, which makes the numbers systematically undercount and hard to trust.
2. **Scope.** GTM4WP's job is to load the container and populate the dataLayer; running a separate blocked-visitor analytics system (its own tab, storage, dashboards) is really a different product concern, and keeping the plugin focused there is deliberate.

If you need a signal on blocked traffic, a server-side tagging setup (server-side GTM / first-party measurement) is the more dependable route, since it isn't defeated by client-side blockers the same way. Closing this as out of scope — but thanks for raising it, and feel free to reopen the discussion if you see it differently.
